helix-core/src/syntax.rs

```diff
@@ -144,34 +142,30 @@ fn read_query(language: &str, filename: &str) -> String {
 
 impl LanguageConfiguration {
     pub fn highlight_config(&self, scopes: &[String]) -> Option<Arc<HighlightConfiguration>> {
-        self.highlight_config
```

Comment:
I think the idea here was to only recalculate `highlight_config` if absolutely necessary (we only need to do this if the theme changed and we have different scopes now). `HighlightConfiguration` isn't cheap to construct and it also takes several `fs` reads so ideally we want to reuse them across files if they're the same language.